### PR TITLE
[keyword](decimalv2) Add DecimalV2 keyword (#26283)

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Pick #26283

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/exec/tablet_info.cpp [hunk 1]
```diff
@@ -92,8 +91,9 @@ Status OlapTableSchemaParam::init(const POlapTableSchemaParam& pschema) {
         for (auto& pcolumn_desc : p_index.columns_desc()) {
             if (!_is_partial_update ||
                 _partial_update_input_columns.count(pcolumn_desc.name()) > 0) {
-                std::string col_type = has_invalid_type ? "INVALID_TYPE" : pcolumn_desc.type();
```
Comment (Lchangliang) [inline be/src/exec/tablet_info.cpp:95]: Can't remove this

File: be/src/exec/tablet_info.cpp [hunk 2]
```diff
@@ -69,19 +69,18 @@ Status OlapTableSchemaParam::init(const POlapTableSchemaParam& pschema) {
...
-    std::unordered_map<std::pair<std::string, std::string>, SlotDescriptor*> slots_map;
+    std::unordered_map<std::pair<std::string, FieldType>, SlotDescriptor*> slots_map;
     _tuple_desc = _obj_pool.add(new TupleDescriptor(pschema.tuple_desc()));
     // When FE version is less than 2.0.3, But BE upgrade to 2.0.3,
     // the filed col_type in slot_desc is INVALID_TYPE default.
     // It can be remove later.
-    bool has_invalid_type = false;
     for (auto& p_slot_desc : pschema.slot_descs()) {
         auto slot_desc = _obj_pool.add(new SlotDescriptor(p_slot_desc));
-        if (slot_desc->col_type() == INVALID_TYPE) has_invalid_type = true;
```
Comment (Lchangliang) [inline be/src/exec/tablet_info.cpp:80]: Can't remove this

**Review by Lchangliang — CHANGES_REQUESTED** [review]

**Comment by Gabriel39** [comment]
run buildall

**Comment by xiaokang** [comment]
run buildall

**Pull request merged**